src/Serilog/Settings/KeyValuePairs/SettingValueConversions.cs

```diff
@@ -114,12 +114,30 @@ public static object ConvertToType(string value, Type toType)
             if (toType == typeof(String[]))
             {
                 if (string.IsNullOrEmpty(value))
-                    return new String[] { };
+                    return new String[] {};
 
                 var items = value.Split(',');
                 return items;
             }
 
+            if (toType == typeof(int[]))
+            {
+                if (string.IsNullOrEmpty(value))
+                    return new int[] {};
+
+                var list = new List<int>();
+                var items = value.Split(',');
+                foreach(var item in items)
+                {
+                    if(int.TryParse(item, out int result))
+                    {
+                        list.Add(result);
+                    }
+
+                }
+                return list.ToArray();
+            }
+
```

Comment:
Thanks for the update!

This approach might get a bit onerous once we have to support `bool[]`, `long[]`, etc.

Rather than these two blocks, we can support any collection type with:

```csharp
if (toType.IsArray && toType.GetArrayRank() == 1)
{
    var elementType = toType.GetElementType();
    var items = value?.Split(',');
    var length = items?.Length ?? 0;
    var result = Array.CreateInstance(elementType, length);
    if (items != null)
    {
        for (var i = 0; i < length; ++i)
        {
            var item = items[i];
            var element = ConvertToType(item, elementType);
            result.SetValue(element, i);
        }
    }
    return result;
}
```

(Should keep the existing and added tests passing 🤞 )

Reply:
Implemented your code and fixed failing tests on empty arrays. I think it's good to be merged into dev?